presto-main/src/main/java/io/prestosql/metadata/MetadataManager.java

```diff
@@ -1099,10 +1099,16 @@ public void validateScan(Session session, TableHandle table)
 
         ConnectorSession connectorSession = session.toConnectorSession(catalogName);
         return metadata.applyProjection(connectorSession, table.getConnectorHandle(), projections, assignments)
-                .map(result -> new ProjectionApplicationResult<>(
-                        new TableHandle(catalogName, result.getHandle(), table.getTransaction(), Optional.empty()),
-                        result.getProjections(),
-                        result.getAssignments()));
+                .map(result -> {
+                    // verify no nulls
+                    List<ConnectorExpression> newProjections = ImmutableList.copyOf(result.getProjections());
+                    List<ProjectionApplicationResult.Assignment> newAssignments = ImmutableList.copyOf(result.getAssignments());
```

Comment:
Be explicit about it (and throw a proper exception):

```
if (!result.getProjections().stream().allMatch(Objects::isNull)) {
    throw new PrestoException(...);
}

if (!result.getAssignments().stream().allMatch(Objects::isNull)) {
    throw new PrestoException(...);
}
```

Reply:
changed to explicit code (but slightly different)